griptape/engines/image_generation/image_generation_engine.py

```diff
@@ -0,0 +1,10 @@
+from attr import field, define
+from griptape.drivers import BaseImageGenerationDriver
+
+
+@define
+class ImageGenerationEngine:
+    image_generation_driver: BaseImageGenerationDriver = field(kw_only=True)
+
+    def generate_image(self, prompts: list[str], negative_prompts: list[str], **kwargs):
+        return self.image_generation_driver.generate_image(prompts, negative_prompts, **kwargs)
```

Comment:
I think the Engine should use `J2` to render out a single string containing prompts, negative prompts, rules, etc. Then this could be passed to the driver.

Reply:
Curious what this would look like, maybe you can expand on that idea? The drivers and their underlying models, as a rule, provide very heterogenous interfaces (see ChatGPT vs. Stable Diffusion prompt discussion [on Slack](https://griptape-ai.slack.com/archives/C05T3HLBH52/p1698955113291039)). Would this mean the engine renders a prompt string which then has to be parsed by the driver and passed to the model as API parameters?

More concretely: DALLE2/3 accepts a single prompt string (not sure if it can support negative weighting in prompts, could be experimented with), Leonardo expects a single prompt and a single negative prompt string, and Stable Diffusion accepts an arbitrary list of prompts with user-defined weights [-1, 1].